cmd/kubeadm/app/discovery/token/token.go

```diff
@@ -206,7 +218,7 @@ func runForEndpointsAndReturnFirst(endpoints []string, fetchKubeConfigFunc func(
 		}(endpoint)
 	}
 	wg.Wait()
-	return resultingKubeConfig
+	return resultingKubeConfig, resultingError
```

Comment:
`resultingError` can be mutated if we've got multiple endpoints, right?

If so, I prefer not to return an extra error, which will cause potential issues. For example,  if one of the endpoints is accessible, in this case, we should return an available kubeconfig with nil error, right? Then what we really need to do is cautious about nil kubeconfig when using it.

Reply:
Yes, you are quite right there. Somehow I missed that (probably because I test with a single master, hence a single endpoint)